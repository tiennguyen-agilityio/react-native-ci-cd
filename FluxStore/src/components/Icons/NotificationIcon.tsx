import {memo} from 'react';
import {TouchableOpacity} from 'react-native';
import Svg, {Circle, Path} from 'react-native-svg';

import {IconProps} from '@/interfaces';
import {useThemeStore} from '@/stores';
import {colors} from '@/themes';

interface NotificationIconProps extends IconProps {
  isNews?: boolean;
}

export const NotificationIcon = memo(
  ({
    width = 26,
    height = 26,
    color,
    disabled = false,
    isNews = false,
    style,
    onPress,
  }: NotificationIconProps) => {
    const {
      theme: {icon},
    } = useThemeStore();

    return (
      <TouchableOpacity
        testID="notification-icon"
        activeOpacity={0.8}
        onPress={onPress}
        disabled={disabled}
        style={style}>
        <Svg width={width} height={height} viewBox="0 0 26 26" fill="none">
          {isNews ? (
            <>
              <Path
                fillRule="evenodd"
                clipRule="evenodd"
                d="M15.662 3.041a6.802 6.802 0 00-9.422 5.51l-.273 2.455-.039.333a7.25 7.25 0 01-1.06 2.924l-.627 1.044-.024.04c-.298.497-.557.928-.718 1.289-.165.368-.304.82-.154 1.302.075.238.199.457.364.643.336.376.796.49 1.196.538.393.048.895.048 1.475.048H19.62c.58 0 1.083 0 1.475-.048.4-.048.86-.162 1.196-.538.166-.186.29-.405.364-.643.15-.481.011-.934-.153-1.302-.162-.361-.42-.792-.719-1.29l-.023-.04-.627-1.043a7.247 7.247 0 01-1.016-2.633 5.21 5.21 0 01-1.524-.043 8.752 8.752 0 001.249 3.439l.005.009.626 1.044c.33.548.54.9.66 1.169.07.158.083.232.086.256a.25.25 0 01-.039.068.925.925 0 01-.264.058c-.292.035-.702.037-1.342.037H6.426c-.64 0-1.05-.002-1.341-.037a.927.927 0 01-.265-.058.25.25 0 01-.038-.067.926.926 0 01.086-.257c.12-.268.33-.62.66-1.17l.625-1.043.005-.008c.086-.144.136-.226.183-.309a8.75 8.75 0 001.116-3.537v-.01l.273-2.455a5.302 5.302 0 017.085-4.397c.218-.468.505-.898.847-1.278zm2.46 1.786c-.424.35-.714.86-.776 1.437.404.578.697 1.242.846 1.964a2.157 2.157 0 001.58.422l-.012-.1a6.78 6.78 0 00-1.637-3.723zm3.096 12.684v0zm-.03.057l-.003.001.002-.001zm-16.376 0zm-.03-.054v-.003.003z"
                fill={color ?? icon.primary}
              />
              <Path
                d="M9.86 19.137c.186 1.036.594 1.952 1.162 2.606.567.653 1.263 1.007 1.978 1.007.715 0 1.41-.354 1.979-1.007.567-.654.975-1.57 1.16-2.606"
                stroke={color ?? icon.primary}
                strokeWidth={1.5}
                strokeLinecap="round"
              />
              <Circle cx={18.5} cy={7.5} r={3.5} fill={colors.pink[900]} />
            </>
          ) : (
            <>
              <Path
                d="M13 2.5a6.802 6.802 0 016.76 6.05l.273 2.456c.02.178.028.256.038.333a7.25 7.25 0 00.89 2.636l.172.289.626 1.043c.308.514.576.958.742 1.329.164.368.303.821.153 1.303a1.754 1.754 0 01-.364.642c-.335.376-.795.49-1.195.538-.403.049-.922.048-1.522.048H6.427c-.6 0-1.12 0-1.522-.048-.4-.048-.86-.162-1.195-.538a1.753 1.753 0 01-.365-.642c-.15-.482-.01-.935.154-1.303.165-.37.434-.815.742-1.33l.626-1.042c.092-.154.133-.222.17-.29a7.25 7.25 0 00.891-2.635l.039-.333.273-2.455A6.801 6.801 0 0113 2.5zM13 4a5.302 5.302 0 00-5.27 4.717l-.272 2.455c-.02.172-.03.269-.043.365A8.753 8.753 0 016.34 14.72l-.188.316-.626 1.044c-.329.548-.54.901-.66 1.169a.928.928 0 00-.085.256.25.25 0 00.04.067c.021.01.092.038.263.059.291.035.702.037 1.342.037h13.146c.64 0 1.05-.002 1.342-.037a.913.913 0 00.262-.059.246.246 0 00.04-.067.909.909 0 00-.085-.256c-.12-.268-.33-.62-.66-1.169l-.626-1.044c-.088-.148-.139-.232-.187-.316a8.755 8.755 0 01-1.117-3.547l-.273-2.455A5.303 5.303 0 0013 4z"
                fill={color ?? icon.primary}
              />
              <Path
                d="M9.86 19.137c.186 1.036.594 1.952 1.162 2.606.567.653 1.263 1.007 1.978 1.007.715 0 1.41-.354 1.979-1.007.567-.654.975-1.57 1.16-2.606"
                stroke={color ?? icon.primary}
                strokeWidth={1.5}
                strokeLinecap="round"
              />
            </>
          )}
        </Svg>
      </TouchableOpacity>
    );
  },
);
