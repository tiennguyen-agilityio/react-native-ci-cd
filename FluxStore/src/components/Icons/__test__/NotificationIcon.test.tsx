import {render, fireEvent, screen} from '@testing-library/react-native';

import {NotificationIcon} from '../NotificationIcon';

describe('NotificationIcon', () => {
  const onPress = jest.fn();

  it('renders correctly (matches snapshot)', () => {
    const {toJSON} = render(<NotificationIcon />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly (matches snapshot) with isActive', () => {
    const {toJSON} = render(<NotificationIcon isActive />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly (matches snapshot) with isNews', () => {
    const {toJSON} = render(<NotificationIcon isNews />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('calls onPress when touched and not disabled', () => {
    render(<NotificationIcon onPress={onPress} />);

    fireEvent.press(screen.getByTestId('notification-icon'));

    expect(onPress).toHaveBeenCalledTimes(1);
  });

  it('does NOT call onPress when disabled', () => {
    render(<NotificationIcon onPress={onPress} disabled />);

    fireEvent.press(screen.getByTestId('notification-icon'));

    expect(onPress).not.toHaveBeenCalledTimes(0);
  });
});
