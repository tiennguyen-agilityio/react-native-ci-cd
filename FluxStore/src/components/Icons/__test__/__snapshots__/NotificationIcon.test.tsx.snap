// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationIcon renders correctly (matches snapshot) 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="notification-icon"
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={26}
    bbWidth={26}
    fill="none"
    focusable={false}
    height={26}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 26,
          "width": 26,
        },
      ]
    }
    vbHeight={26}
    vbWidth={26}
    width={26}
  >
    <RNSVGGroup
      fill={null}
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGPath
        d="M13 2.5a6.802 6.802 0 016.76 6.05l.273 2.456c.02.178.028.256.038.333a7.25 7.25 0 00.89 2.636l.172.289.626 1.043c.308.514.576.958.742 1.329.164.368.303.821.153 1.303a1.754 1.754 0 01-.364.642c-.335.376-.795.49-1.195.538-.403.049-.922.048-1.522.048H6.427c-.6 0-1.12 0-1.522-.048-.4-.048-.86-.162-1.195-.538a1.753 1.753 0 01-.365-.642c-.15-.482-.01-.935.154-1.303.165-.37.434-.815.742-1.33l.626-1.042c.092-.154.133-.222.17-.29a7.25 7.25 0 00.891-2.635l.039-.333.273-2.455A6.801 6.801 0 0113 2.5zM13 4a5.302 5.302 0 00-5.27 4.717l-.272 2.455c-.02.172-.03.269-.043.365A8.753 8.753 0 016.34 14.72l-.188.316-.626 1.044c-.329.548-.54.901-.66 1.169a.928.928 0 00-.085.256.25.25 0 00.04.067c.021.01.092.038.263.059.291.035.702.037 1.342.037h13.146c.64 0 1.05-.002 1.342-.037a.913.913 0 00.262-.059.246.246 0 00.04-.067.909.909 0 00-.085-.256c-.12-.268-.33-.62-.66-1.169l-.626-1.044c-.088-.148-.139-.232-.187-.316a8.755 8.755 0 01-1.117-3.547l-.273-2.455A5.303 5.303 0 0013 4z"
        fill={
          {
            "payload": 4281544750,
            "type": 0,
          }
        }
        propList={
          [
            "fill",
          ]
        }
      />
      <RNSVGPath
        d="M9.86 19.137c.186 1.036.594 1.952 1.162 2.606.567.653 1.263 1.007 1.978 1.007.715 0 1.41-.354 1.979-1.007.567-.654.975-1.57 1.16-2.606"
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
        propList={
          [
            "stroke",
            "strokeWidth",
            "strokeLinecap",
          ]
        }
        stroke={
          {
            "payload": 4281544750,
            "type": 0,
          }
        }
        strokeLinecap={1}
        strokeWidth={1.5}
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;

exports[`NotificationIcon renders correctly (matches snapshot) with isActive 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="notification-icon"
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={26}
    bbWidth={26}
    fill="none"
    focusable={false}
    height={26}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 26,
          "width": 26,
        },
      ]
    }
    vbHeight={26}
    vbWidth={26}
    width={26}
  >
    <RNSVGGroup
      fill={null}
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGPath
        d="M13 2.5a6.802 6.802 0 016.76 6.05l.273 2.456c.02.178.028.256.038.333a7.25 7.25 0 00.89 2.636l.172.289.626 1.043c.308.514.576.958.742 1.329.164.368.303.821.153 1.303a1.754 1.754 0 01-.364.642c-.335.376-.795.49-1.195.538-.403.049-.922.048-1.522.048H6.427c-.6 0-1.12 0-1.522-.048-.4-.048-.86-.162-1.195-.538a1.753 1.753 0 01-.365-.642c-.15-.482-.01-.935.154-1.303.165-.37.434-.815.742-1.33l.626-1.042c.092-.154.133-.222.17-.29a7.25 7.25 0 00.891-2.635l.039-.333.273-2.455A6.801 6.801 0 0113 2.5zM13 4a5.302 5.302 0 00-5.27 4.717l-.272 2.455c-.02.172-.03.269-.043.365A8.753 8.753 0 016.34 14.72l-.188.316-.626 1.044c-.329.548-.54.901-.66 1.169a.928.928 0 00-.085.256.25.25 0 00.04.067c.021.01.092.038.263.059.291.035.702.037 1.342.037h13.146c.64 0 1.05-.002 1.342-.037a.913.913 0 00.262-.059.246.246 0 00.04-.067.909.909 0 00-.085-.256c-.12-.268-.33-.62-.66-1.169l-.626-1.044c-.088-.148-.139-.232-.187-.316a8.755 8.755 0 01-1.117-3.547l-.273-2.455A5.303 5.303 0 0013 4z"
        fill={
          {
            "payload": 4281544750,
            "type": 0,
          }
        }
        propList={
          [
            "fill",
          ]
        }
      />
      <RNSVGPath
        d="M9.86 19.137c.186 1.036.594 1.952 1.162 2.606.567.653 1.263 1.007 1.978 1.007.715 0 1.41-.354 1.979-1.007.567-.654.975-1.57 1.16-2.606"
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
        propList={
          [
            "stroke",
            "strokeWidth",
            "strokeLinecap",
          ]
        }
        stroke={
          {
            "payload": 4281544750,
            "type": 0,
          }
        }
        strokeLinecap={1}
        strokeWidth={1.5}
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;

exports[`NotificationIcon renders correctly (matches snapshot) with isNews 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="notification-icon"
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={26}
    bbWidth={26}
    fill="none"
    focusable={false}
    height={26}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 26,
          "width": 26,
        },
      ]
    }
    vbHeight={26}
    vbWidth={26}
    width={26}
  >
    <RNSVGGroup
      fill={null}
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGPath
        clipRule={0}
        d="M15.662 3.041a6.802 6.802 0 00-9.422 5.51l-.273 2.455-.039.333a7.25 7.25 0 01-1.06 2.924l-.627 1.044-.024.04c-.298.497-.557.928-.718 1.289-.165.368-.304.82-.154 1.302.075.238.199.457.364.643.336.376.796.49 1.196.538.393.048.895.048 1.475.048H19.62c.58 0 1.083 0 1.475-.048.4-.048.86-.162 1.196-.538.166-.186.29-.405.364-.643.15-.481.011-.934-.153-1.302-.162-.361-.42-.792-.719-1.29l-.023-.04-.627-1.043a7.247 7.247 0 01-1.016-2.633 5.21 5.21 0 01-1.524-.043 8.752 8.752 0 001.249 3.439l.005.009.626 1.044c.33.548.54.9.66 1.169.07.158.083.232.086.256a.25.25 0 01-.039.068.925.925 0 01-.264.058c-.292.035-.702.037-1.342.037H6.426c-.64 0-1.05-.002-1.341-.037a.927.927 0 01-.265-.058.25.25 0 01-.038-.067.926.926 0 01.086-.257c.12-.268.33-.62.66-1.17l.625-1.043.005-.008c.086-.144.136-.226.183-.309a8.75 8.75 0 001.116-3.537v-.01l.273-2.455a5.302 5.302 0 017.085-4.397c.218-.468.505-.898.847-1.278zm2.46 1.786c-.424.35-.714.86-.776 1.437.404.578.697 1.242.846 1.964a2.157 2.157 0 001.58.422l-.012-.1a6.78 6.78 0 00-1.637-3.723zm3.096 12.684v0zm-.03.057l-.003.001.002-.001zm-16.376 0zm-.03-.054v-.003.003z"
        fill={
          {
            "payload": 4281544750,
            "type": 0,
          }
        }
        fillRule={0}
        propList={
          [
            "fill",
            "fillRule",
          ]
        }
      />
      <RNSVGPath
        d="M9.86 19.137c.186 1.036.594 1.952 1.162 2.606.567.653 1.263 1.007 1.978 1.007.715 0 1.41-.354 1.979-1.007.567-.654.975-1.57 1.16-2.606"
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
        propList={
          [
            "stroke",
            "strokeWidth",
            "strokeLinecap",
          ]
        }
        stroke={
          {
            "payload": 4281544750,
            "type": 0,
          }
        }
        strokeLinecap={1}
        strokeWidth={1.5}
      />
      <RNSVGCircle
        cx={18.5}
        cy={7.5}
        fill={
          {
            "payload": 4293871215,
            "type": 0,
          }
        }
        propList={
          [
            "fill",
          ]
        }
        r={3.5}
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;
